dashboard/app/models/pd/workshop_material_order.rb

```diff
@@ -50,7 +50,7 @@ class WorkshopMaterialOrder < ActiveRecord::Base
     has_one :workshop, class_name: 'Pd::Workshop', through: :enrollment, foreign_key: :pd_workshop_id
 
     validates :enrollment, presence: true, uniqueness: true
-    validates :user, presence: true, uniqueness: true
+    validates :user, presence: true, uniqueness: true, if: -> {user_id_changed?}
```

Comment:
I don't think so. Note [this](https://github.com/code-dot-org/code-dot-org/blob/885d2d43cfca009506590324f56a06318f89cf88/dashboard/test/models/pd/workshop_material_order_test.rb#L53-L53) `User is required` error is still occurring on a new model. I can add an explicit test for the null user to be sure.